### DISCUSSION TITLE
redis metadata storage... what command is needed?

### DISCUSSION THREAD

**Discussion by vipsland** [juicedata/juicefs#2659] [General]
redis metadata storage... what command is needed?
i have written customized redis server protocol but didnt complete everything.
just curious what command is needed for the metadata storage to be used?

set / get <- anything else?

**Comment by holygrolli** [comment]
I don't get your question. What (exactly) did you try and what (format, mount, ...) do you want to achieve?

**Reply by solracsf** [reply]
I believe the question is about [redis commands](https://redis.io/commands/) metadata uses (and therefore must be whitelisted when using redis ACL). I'm interested too,

**Comment by davies** [comment]
At the top of `meta/redis.go`, we list all the required Redis features:
```
	Redis features:
	  Sorted Set: 1.2+
	  Hash Set: 4.0+
	  Transaction: 2.2+
	  Scripting: 2.6+
	  Scan: 2.8+
```

For the list of all required commends, you need to read the source code.

**Comment by m2acgi** [comment]
@davies , Is this ACL enough for run juicefs `user juicefs-user on >juicefs-password ~* &* +@all -@dangerous` ?
